api.go

```diff
@@ -51,21 +51,21 @@ func CreatePod(podConfig PodConfig) (*Pod, error) {
 	}
 
 	// Initialize the network.
-	err = p.network.init(&(p.config.NetworkConfig))
+	netNsPath, netNsCreated, err := p.network.init(p.config.NetworkConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	// Execute prestart hooks inside netns
-	err = p.network.run(p.config.NetworkConfig.NetNSPath, func() error {
+	err = p.network.run(netNsPath, func() error {
```

Comment:
@sboeuf The pre-start hooks do not need to be executed from within the namespace. At least with docker the pid of the ns is sent as a parameter. So do we need to run within the namespace?